### PR TITLE
Tikz figure export

### PR DESCRIPTION
**Pull request by d-monnet**

### CODE REVIEW

File: src/tikz_export.jl [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+* `solvernames :: Vector{String} = []` : names of the solvers, should have as many elements as the number of columns of `T`. If empty, use the labels returned by `performance_profile_axis_labels`.
+* `xlim::AbstractFloat=10.` : size of the figure along the x axis. /!\\ the legend is added on the right hand side of the figure.
+* `ylim::AbstractFloat=10.` : size of the figure along the y axis.
+* `nxgrad::Int=5` : number of graduations on the x axis.
+* `nygrad::Int=5` : number of graduations on the y axis.
+* `grid::Bool=true` : display grid if true.
+* `colours::Vector{String} = []` : colours of the plots, should have as many elements as the number of columns of `T`.
+* `linestyles::Vector{String} = []` : line style (dashed, dotted, ...) of the plots, should have as many elements as the number of columns of `T`.
+* `linewidth::AbstractFloat = 1.0` : line with of the plots.
```
Comment (tmigot) [inline src/tikz_export.jl:24]:
```suggestion
* `linewidth::AbstractFloat = 1.0` : line width of the plots.
```

File: src/tikz_export.jl [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+* `xlabel::String = ""` : x axis label. If empty, uses the one returns by `performance_profile_axis_labels`.
```
Comment (tmigot) [inline src/tikz_export.jl:25]:
```suggestion
* `xlabel::String = ""` : x-axis label. If empty, uses the one returned by `performance_profile_axis_labels`.
```

File: src/tikz_export.jl [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+* `ylabel::String = ""` : x axis label. If empty, uses the one returns by `performance_profile_axis_labels`.
```
Comment (tmigot) [inline src/tikz_export.jl:26]:
```suggestion
* `ylabel::String = ""` : y-axis label. If empty, uses the one returned by `performance_profile_axis_labels`.
```

File: src/tikz_export.jl [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+Other keyword arguments are passed `performance_profile_data`.
```
Comment (tmigot) [inline src/tikz_export.jl:28]:
```suggestion
Other keyword arguments are passed to `performance_profile_data`.
```

File: src/tikz_export.jl [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  xlabel::String = "",
+  ylabel::String = "",
+  kwargs...)
+
+
+
+  logscale = true
+  if haskey(kwargs,:logscale)
+    logscale = kwargs[:logscale]
+  end
```
Comment (tmigot) [inline src/tikz_export.jl:53]: Why not have `logscale` as a keyword argument instead?

File: src/performance_profiles.jl [hunk 6]
```diff
@@ -155,6 +157,23 @@ function performance_profile(
   )
 end
 
+"""
+    function performance_profile_data_mat(T;kwargs...)
+
+Retruns `performance_profile_data` output (vectors) as matrices. Matrices are padded with NaN if necessary.
```
Comment (tmigot) [inline src/performance_profiles.jl:163]:
```suggestion
Returns `performance_profile_data` output (vectors) as matrices. Matrices are padded with NaN if necessary.
```

File: src/performance_profiles.jl [hunk 7]
```diff
@@ -164,14 +183,14 @@ Export a performance profile plot data as .csv file. Profiles data are padded wi
...
 * `T :: Matrix{Float64}`: each column of `T` defines the performance data for a solver (smaller is better).
   Failures on a given problem are represented by a negative value, an infinite value, or `NaN`.
-* `filename :: String` : path to the export file.
+* `filename :: String` : path to the exported file.
 
 ## Keyword Arguments
 
-* `solver_names :: Vector{S}` : names of the solvers
-* `header::Vector{String}`: Contains .csv file column names. Note that `header` value does not change columns order in .csv exported files (see Output).
+* `solver_names :: Vector{S}` : names of the solvers.
```
Comment (tmigot) [inline src/performance_profiles.jl:190]:
```suggestion
- `solver_names :: Vector{S}` : names of the solvers.
```

File: src/tikz_export.jl [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+  # some offsets
+  axis_tik_l = 0.2 # tick length
+  lgd_offset = 0.5 # space between figure and legend box
+  lgd_box_length = 3. # legend box length
+  lgd_plot_length = 0.7 # legend plot length
+  lgd_v_offset = 0.7 # vertical space between legend items 
+
+  label_val = [0.2,0.25,0.5,1] # possible graduation labels along axes are multiples of label_val elements times 10^n
+  ymax = 1.0
```
Comment (tmigot) [inline src/tikz_export.jl:66]: Should these be keyword arguments?

**Review by tmigot — CHANGES_REQUESTED** [review]
This looks overall good for me, just a couple of comments.

There was also @dpo comment https://github.com/JuliaSmoothOptimizers/BenchmarkProfiles.jl/pull/104#issuecomment-1782279646

File: src/tikz_export.jl (hunk 8 above)
Comment (d-monnet) [reply src/tikz_export.jl:66]: They can be indeed, I suppose it's better to give the user as much freedom as possible.

**Comment by dpo** [comment]
Have you seen https://github.com/JuliaTeX/TikzPictures.jl ? It also allows exporting the TikZ to PDF, SVG, etc.

**Comment by d-monnet** [comment]
> Have you seen https://github.com/JuliaTeX/TikzPictures.jl ? It also allows exporting the TikZ to PDF, SVG, etc.

Yes, I've just tried it and it works great ! I'll add different export options to this PR based on that library.

**Comment by d-monnet** [comment]
Last commit implements `TikzPictures.jl` interface to export in different formats. If you have spare time you can generate a few .pdf and suggest improvements/ additional features.

File: src/tikz_export.jl (hunk 5 above)
Comment (tmigot) [reply src/tikz_export.jl:68]: @d-monnet you missed this probably

File: Project.toml [hunk 9]
```diff
@@ -9,14 +9,15 @@ NaNMath = "77ba4419-2d1f-58cd-9bb1-8ffee604a2e3"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 Requires = "ae029012-a4dd-5104-9daa-d747884805df"
 Tables = "bd369af6-aec1-5ad0-b16a-f7cc5008161c"
+TikzPictures = "37f6aa50-8035-52d0-81c2-5a1d08754b2d"
```
Comment (tmigot) [inline Project.toml:12]: You need to also add this to the compat section

**Review by tmigot — CHANGES_REQUESTED** [review]
Thanks @d-monnet for the changes. I have a couple more comments.

After this PR, I would suggest to open a few issues for developing this:
- add this new feature in the package documentation
- use Requires.jl or similar to add TikzPictures.jl as an optional dep.

Surely, this will be improved as we get more practical cases.

File: src/tikz_export.jl (hunk 5 above)
Comment (d-monnet) [reply src/tikz_export.jl:68]: `logscale` is already included in the kwargs that are passed to `performance_profile_data`.

**Comment by d-monnet** [comment]
> Thanks @d-monnet for the changes. I have a couple more comments.
> 
> After this PR, I would suggest to open a few issues for developing this:
> 
>     * add this new feature in the package documentation
> 
>     * use Requires.jl or similar to add TikzPictures.jl as an optional dep.
> 
> 
> Surely, this will be improved as we get more practical cases.

Sure, I'll open the issues once you confirm that the PR can be merged.

File: src/tikz_export.jl (hunk 5 above)
Comment (tmigot) [reply src/tikz_export.jl:68]: You can add in the keyword arguments of thé function and, then, when calling `performance_profile_data` you add `logscale = logscale`.
You construction seems unnecessary to me.

Comment (d-monnet) [reply src/tikz_export.jl]: You're wright, not sure what I was thinking about... This is fixed with the last commit !

**Review by tmigot — APPROVED** [review]
LGTM

**Comment by d-monnet** [comment]
> LGTM, thanks!

Thanks for the review @tmigot !

**Pull request merged**